executor/src/witgen/rows.rs

```diff
@@ -0,0 +1,315 @@
+use std::fmt::Debug;
+
+use ast::analyzed::{Expression, PolynomialReference, PolynomialType};
+use itertools::Itertools;
+use number::{DegreeType, FieldElement};
+
+use crate::witgen::Constraint;
+
+use super::{
+    affine_expression::{AffineExpression, AffineResult},
+    column_map::ColumnMap,
+    expression_evaluator::ExpressionEvaluator,
+    global_constraints::RangeConstraintSet,
+    range_constraints::RangeConstraint,
+    symbolic_witness_evaluator::{SymoblicWitnessEvaluator, WitnessColumnEvaluator},
+    EvalValue, FixedData,
+};
+
+/// A single cell, holding an optional value and range constraint.
+#[derive(Clone)]
+pub struct Cell<T: FieldElement> {
+    /// The column name, for debugging purposes.
+    pub name: &'static str,
+    /// The value of the cell, if known.
+    pub value: Option<T>,
+    /// Range constraints of the cell, if any.
+    pub range_constraint: Option<RangeConstraint<T>>,
```

Comment:
I think this would be cleaner as a single enum. Fine for this PR if it's a big change.

Reply:
Not a huge change, and much better! Now cells also get smaller, because they can't have both a value and a range constraint 🎉